### PR TITLE
feat(config): Add connection indicators flags.

### PR DESCRIPTION
**Pull request by tudor-phd**

### CODE REVIEW

File: react/features/connection-indicator/components/AbstractConnectionIndicator.js [hunk 1]
```diff
@@ -165,9 +172,26 @@ export default class AbstractConnectionIndicator<P: Props, S: State> extends Com
...
 }
+
+/**
+ * Maps (parts of) the Redux state to the associated props for the
+ * {@code ConnectorIndicator} component.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _autoHideTimeout: number
```
Comment (saghul) [inline react/features/connection-indicator/components/AbstractConnectionIndicator.js:187]: just say `{Props}` here.

File: react/features/connection-indicator/components/AbstractConnectionIndicator.js [hunk 2]
```diff
@@ -165,9 +172,26 @@ export default class AbstractConnectionIndicator<P: Props, S: State> extends Com
...
+ * }}
+ */
+export function mapStateToProps(state: Object) {
+
```
Comment (saghul) [inline react/features/connection-indicator/components/AbstractConnectionIndicator.js:191]: please remove the unneeded newline

File: react/features/filmstrip/components/web/Thumbnail.js [hunk 3]
```diff
@@ -1093,8 +1093,10 @@ function _mapStateToProps(state, ownProps): Object {
 
     return {
         _audioTrack,
-        _connectionIndicatorAutoHideEnabled: interfaceConfig.CONNECTION_INDICATOR_AUTO_HIDE_ENABLED,
-        _connectionIndicatorDisabled: _isMobile || interfaceConfig.CONNECTION_INDICATOR_DISABLED,
+        _connectionIndicatorAutoHideEnabled:
+        Boolean(state['features/base/config'].connectionIndicators?.autoHide ?? true),
+        _connectionIndicatorDisabled: _isMobile
+        || Boolean(state['features/base/config'].connectionIndicators?.disabled),
```
Comment (saghul) [inline react/features/filmstrip/components/web/Thumbnail.js:1099]: can you indent this one more level to make it easier to read?

File: react/features/base/config/reducer.js [hunk 4]
```diff
@@ -194,6 +194,17 @@ function _translateLegacyConfig(oldValue: Object) {
         newValue.toolbarButtons = interfaceConfig.TOOLBAR_BUTTONS;
     }
 
+    if (!oldValue.connectionIndicators
+    && typeof interfaceConfig === 'object' && (interfaceConfig.CONNECTION_INDICATOR_DISABLED
+        || interfaceConfig.CONNECTION_INDICATOR_AUTO_HIDE_ENABLED
```
Comment (saghul) [inline react/features/base/config/reducer.js:199]: I think we need to checck for these differently sine they are boolean. We need to check if the object contains those properties, not their values.

File: react/features/base/config/reducer.js [hunk 5]
```diff
@@ -194,6 +194,17 @@ function _translateLegacyConfig(oldValue: Object) {
         newValue.toolbarButtons = interfaceConfig.TOOLBAR_BUTTONS;
     }
 
+    if (!oldValue.connectionIndicators
+    && typeof interfaceConfig === 'object' && (interfaceConfig.CONNECTION_INDICATOR_DISABLED
```
Comment (saghul) [inline react/features/base/config/reducer.js:198]: please indent this one more level

File: config.js [hunk 6]
```diff
@@ -252,6 +252,14 @@ var config = {
     // Default value for the channel "last N" attribute. -1 for unlimited.
     channelLastN: -1,
 
+    // Connection indicators
+    // connectionIndicators: {
```
Comment (saghul) [inline config.js:256]: Please remove the legacy options from interface_config.js . Just leave a comment saying they got moved to config.js.

**Review by saghul — CHANGES_REQUESTED** [review]
Left some ccomments, PTAL.

File: interface_config.js [hunk 7]
```diff
@@ -25,31 +25,8 @@ var interfaceConfig = {
...
-     */
-    CONNECTION_INDICATOR_AUTO_HIDE_TIMEOUT: 5000,
-
-    /**
-     * If true, hides the connection indicators completely.
-     *
-     * @type {boolean}
-     */
-    CONNECTION_INDICATOR_DISABLED: false,
+    // Connection indicators got moved to config.js
```
Comment (saghul) [inline interface_config.js:29]: Please keep the names of the options listed, so when one greps for them you can see them there.

File: react/features/connection-indicator/components/web/ConnectionIndicator.js [hunk 8]
```diff
@@ -225,7 +230,8 @@ class ConnectionIndicator extends AbstractConnectionIndicator<Props, AbstractSta
      */
     _renderIcon() {
         if (this.props._connectionStatus
-            === JitsiParticipantConnectionStatus.INACTIVE) {
+            === JitsiParticipantConnectionStatus.INACTIVE
+            && !this.props._connectionIndicatorInactiveDisabled) {
```
Comment (saghul) [inline react/features/connection-indicator/components/web/ConnectionIndicator.js:234]: Is this the right place to do this? IIRC there is another place which checks if an icon has to be rendered at all. Can you check?

**Review by saghul — CHANGES_REQUESTED** [review]

File: interface_config.js [hunk 9]
```diff
@@ -25,31 +25,8 @@ var interfaceConfig = {
...
-     */
-    CONNECTION_INDICATOR_AUTO_HIDE_TIMEOUT: 5000,
-
-    /**
-     * If true, hides the connection indicators completely.
-     *
-     * @type {boolean}
-     */
-    CONNECTION_INDICATOR_DISABLED: false,
+    // Connection indicators (CONNECTION_INDICATOR_AUTO_HIDE_ENABLED, CONNECTION_INDICATOR_AUTO_HIDE_TIMEOUT, CONNECTION_INDICATOR_DISABLED) got moved to config.js.
```
Comment (saghul) [inline interface_config.js:29]: Please list them vertically to avoid having such a long line.

**Review by saghul — COMMENTED** [review]
LGTM with a nit.

**Review by saghul — APPROVED** [review]

**Comment by Mohamed-Gibreel** [comment]
Is this working on mobile ? I cannot seem to get any indicator on mobile, how do I enable it ?

**Comment by saghul** [comment]
Not all indicators are available on mobile. It follows a different logic.

**Pull request merged**